test/config-luatest/rpc_test.lua

```diff
@@ -419,6 +419,91 @@ g.test_filter_mode = function(g)
     g.server_4:exec(check)
 end
 
+g.test_filter_works_in_parallel = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+    local config = [[
+    credentials:
+      users:
+        guest:
+          roles: [super]
+        myuser:
+          password: "secret"
+          roles: [replication]
+          privileges:
+          - permissions: [execute]
+            universe: true
+
+    iproto:
+      listen:
+        - uri: 'unix/:./{{ instance_name }}.iproto'
+      advertise:
+        peer:
+          login: 'myuser'
+
+    groups:
+      group-001:
+        replicasets:
+          replicaset-001:
+            instances:
+              instance-001:
+                database:
+                  mode: rw
+          replicaset-002:
+            instances:
+              instance-002:
+                iproto:
+                  listen:
+                    - uri: 1.0.0.0:8888
+              instance-003:
+                iproto:
+                  listen:
+                    - uri: 2.0.0.0:8888
+              instance-004:
+                iproto:
+                  listen:
+                    - uri: 3.0.0.0:8888
+              instance-005:
+                iproto:
+                  listen:
+                    - uri: 4.0.0.0:8888
+    ]]
+    treegen.write_file(dir, 'config.yaml', config)
+
+    local opts = {
+        env = {LUA_PATH = os.environ()['LUA_PATH']},
+        config_file = 'config.yaml',
+        chdir = dir,
+    }
+    g.server_1 = server:new(fun.chain(opts, {alias = 'instance-001'}):tomap())
+
+    g.server_1:start({wait_until_ready = false})
+    -- Intentionally ignore other instances.
+
+    g.server_1:wait_until_ready()
+
+    local function check_filter()
+        local connpool = require('experimental.connpool')
+        local clock = require('clock')
+
+        -- The connection timeout for filter() is hardcoded in
+        -- connpool and equals to 10 seconds.
+        local CONNECT_TIMEOUT = 10
+
+        -- If the connection pool tries to connect in parallel, the
+        -- execution time is bounded with CONNECT_TIMEOUT plus some
+        -- small overhead.
+        --
+        -- Make sure we're trying to access the instances in parallel.
+        local opts = { mode = 'ro' }
+        local timestamp_before_filter = clock.monotonic()
+        t.assert_equals(connpool.filter(opts), {})
```

Comment:
I think it would be good to have a similar test for `call` too -- at least the task mentions that the same speedup effect should occur for `call` too :)